wrangler-service/src/main/java/io/cdap/wrangler/service/adls/ADLSHandler.java

```diff
@@ -0,0 +1,382 @@
+            }
+        });
+    }
+
+    /**
+     * Reads ADLS file into workspace
+     *
+     * @param request   HTTP Request handler.
+     * @param responder HTTP Response handler.
+     */
+    @POST
+    @ReadWrite
+    @Path("contexts/{context}/connections/{connection-id}/adls/read")
+    public void loadADLSFile(HttpServiceRequest request, HttpServiceResponder responder,
+                             @PathParam("context") String namespace,
+                             @PathParam("connection-id") String connectionId,
+                             @QueryParam("path") String filePath, @QueryParam("lines") int lines,
+                             @QueryParam("sampler") String sampler, @QueryParam("fraction") double fraction,
+                             @QueryParam("scope") String scope) {
+        if (Strings.isNullOrEmpty(scope)) {
+            scope = WorkspaceDataset.DEFAULT_SCOPE;
+        }
+        final String scopeS = scope;
+        respond(request, responder, namespace, ns -> {
+            try {
+                if (Strings.isNullOrEmpty(connectionId)) {
+                    throw new BadRequestException("Required path param 'connection-id' is missing in the input");
+                }
+                if (lines == 0) {
+                    throw new NumberFormatException("lines to extract only limit random lines should not be zero");
+                }
+                String header = request.getHeader(PropertyIds.CONTENT_TYPE);
+                NamespacedId namespacedConnId = new NamespacedId(ns, connectionId);
+                Connection connection = getValidatedConnection(namespacedConnId, ConnectionType.ADLS);
+                FileQueryDetails fileQueryDetails = new FileQueryDetails(header, filePath, lines, sampler,
+                        fraction, scopeS);
+                ADLSConnectionSample sample = fetchFileFromClient(connection, fileQueryDetails, namespacedConnId);
+                return new ServiceResponse<>(sample);
+            } catch (ADLException e) {
+                throw new StatusCodeException(e.getMessage(), e, e.httpResponseCode);
+            }
+        });
+    }
+
+    /**
+     * A method to fetch a file from an ADLS client
+     *
+     * @param connection
+     * @param fileQueryDetails
+     * @param namespaceID
+     * @throws IOException
+     */
+    private ADLSConnectionSample fetchFileFromClient(Connection connection, FileQueryDetails fileQueryDetails,
+                                                     NamespacedId namespaceID) throws IOException {
+        ADLStoreClient client = initializeAndGetADLSClient(connection);
+        DirectoryEntry file = ADLSUtility.getFileFromClient(client, fileQueryDetails.getFilePath());
+        try (InputStream inputStream = ADLSUtility.clientInputStream(client, fileQueryDetails)) {
+            if (fileQueryDetails.getHeader() != null && fileQueryDetails.getHeader().equalsIgnoreCase
```

Comment:
this condition can be simplified into `"text/plain".equalsIgnoreCase(fileQueryDetails.getHeader())`